### PR TITLE
Add Mergify for patch updates

### PR DESCRIPTION
**Pull request by angoglez**
This change has been made by @angoglez from https://mergify.io simulator.

---

<details>
<summary>Mergify commands and options</summary>

<br />

More conditions and actions can be found in the [documention](https://doc.mergify.io/).

You can also trigger Mergify actions by commenting on this pull request:

- `@Mergifyio refresh` will re-evaluate the rules
- `@Mergifyio rebase` will rebase this PR
- `@Mergifyio backports <destination>` will backport this PR on `<destination>` branch

Additionally, on Mergify [dashboard](https://dashboard.mergify.io/) you can:

- look at your merge queues
- generate the Mergify configuration with the simulator.

Finally, you can contact us on https://mergify.io/
</details>

### CODE REVIEW

**Comment by JesusMtnez** [comment]
Right now 1 review is required to merge a PR. Will it affect `mergify` prs?

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pull_request_rules:
+  - name: Automatic merge scala-steward PRs when CI is passing
+    conditions:
+      - author=scala-steward
+      - status-success=continuous-integration/travis-ci/pr
+      - body~=labels:.*semver-patch
+    actions:
+      merge:
+        method: merge
```
Comment (JesusMtnez) [inline .mergify.yml]: Not sure if the method is called `squash`

```suggestion
        method: squash
```

Comment (angoglez) [reply .mergify.yml]: Yep, squash can be used like that.

**Comment by angoglez** [comment]
> Right now 1 review is required to merge a PR. Will it affect `mergify` prs?

Honestly, I am not sure about that. If you want to keep following this policy that rule can be added to the config file and then the automatic merge won't  bee done until it has 1 approval, plus the passing CI and the PR's authorhip by scala-steward.

**Pull request merged**